lib/class-wp-rest-post-search-handler.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * REST API: WP_REST_Post_Search_Handler class
+ *
+ * @package gutenberg
+ * @since 3.3.0
+ */
+
+/**
+ * Core class representing a search handler for posts in the REST API.
+ *
+ * @since 3.3.0
+ */
+class WP_REST_Post_Search_Handler extends WP_REST_Object_Search_Handler {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.3.0
+	 */
+	public function __construct() {
+		$this->type     = 'post';
+		$this->subtypes = array_values( get_post_types( array(
+			'public'       => true,
+			'show_in_rest' => true,
+		), 'names' ) );
+	}
+
+	/**
+	 * Searches the object type content for a given search request.
+	 *
+	 * @since 3.3.0
+	 *
+	 * @param WP_REST_Request $request Full REST request.
+	 * @return array Associative array containing an `WP_REST_Object_Search_Handler::RESULT_IDS` containing
+	 *               an array of found IDs and `WP_REST_Object_Search_Handler::RESULT_TOTAL` containing the
+	 *               total count for the matching search results.
+	 */
+	public function search_items( WP_REST_Request $request ) {
```

Comment:
Core doesn't typically type cast.

Reply:
True, but I don't see why this cannot change. To be fair, there are a few areas in core where type hints are present, and them being available (where possible in PHP 5.2) ensures that the parameter is valid.

I'm not strongly opposed to changing it, but only if there's other arguments/opinions against it.

Reply:
There are a handful of uses in Core, I have no problem with adding this one.

(Run `ack --php 'ack --php '(?:^|\s)function [^(]+\([^)]*?(\S+[^(,] \$[^ ),]+)''` on Core for examples.)